### PR TITLE
pipeline for dataset json to augmented dataset

### PR DESCRIPTION
**Pull request by tmabraham**
okay I got a pipeline working that takes the data json and a augmentation config json, duplicates the data, and does the augmentations...

right now it does have a couple issues:
- it seems to augment the original question/answer even thought i had some code to avoid that
- it's toooo slow, we need to implement the multiprocessing stuff...

### CODE REVIEW

**Comment by chadbrewbaker** [comment]
ModuleNotFoundError: No module named 'data.helper'
When running 'python3 data/pipeline.py'

'python3 data/augs.py' is complaining about  FileNotFoundError: [Errno 2] No such file or directory: 'dataset/CodeReviewSE.json'

Fresh clone on M1 OSX homebrew python3 toolchain. Am I doing something stupid?

Where is the dataset json? All I have is @moyix's massive Debian jsonl file.

**Comment by chadbrewbaker** [comment]
# python3 data/augs.py    
dict_keys(['keyboardaug', 'spellingaug', 'backtranslationaug'])
There are: 
153168 questions
100%|█████████████████████████████████████████████████████████████████████████████████████████████████| 76584/76584 [20:06<00:00, 63.47it/s]
Traceback (most recent call last):
   "data/augs.py", line 277, in <module>
    print(data['1']['body'])
TypeError: list indices must be integers or slices, not str

File: data/helper.py [hunk 1]
```diff
@@ -29,14 +43,36 @@ def get_accepted_answer(code_review_data:dict):
...
+            continue
+        # apply augmentation
+        body_strings[i] = augs([body_string])[0]
+    return ' '.join(body_strings)
 
 
 if __name__ == "__main__":
     dataset = load_json_file("dataset/CodeReviewSE.json")
     dataset = dataset[list(dataset.keys())[1000]]
     # print(dataset["body"])
```
Comment (LouisCastricato) [inline data/helper.py:72]: Can you remove the stray comments?

File: data/pipeline.py [hunk 2]
```diff
@@ -29,4 +27,18 @@ def apply_config_prerpoc(self)->dict:
...
+
+    def save_preproc_dataset(self)->None:
+        """
+        Save the preprocessed dataset.
+        """
+        dump_json_file(self.config["output_path"],self.dataset)
+
+
+
+if __name__ == "__main__":
```
Comment (LouisCastricato) [inline data/pipeline.py:41]: Maybe a comment about what this does might be useful? Some sparse documentation.

**Comment by LouisCastricato** [comment]
LGTM

**Pull request merged**